### DISCUSSION TITLE
Does promxy do load balancing also ?

### DISCUSSION THREAD

**Discussion by megarajan** [jacksontj/promxy#630] [General]
we currently have loadbalancer(haproxy) in front of 3 prometheus servers ( all with same scrape config).
we have datasource in grafana pointing the loadbalancer .

we run into holes in grafana whenever a prometheus server is rebooted . 
can I replace my loadbalancer with promxy to avoid this issue ? 
If I do what are the pros / cons ?

will I lose load balancing capabilities ?

**Comment by jacksontj** [comment]
>  can I replace my loadbalancer with promxy to avoid this issue ?

Yes, this  "filling holes" is actually the primary motivation for the creation of this project (https://github.com/jacksontj/promxy/blob/master/MOTIVATION.md).

> will I lose load balancing capabilities ?

That motivation doc covers in a bit more detail -- but you sort-of lose "load balancing" capabilities; because you want a correctly aggregated view of your data -- not a simple load balancer. So promxy understands the API sufficiently to provide that unified interface but because of that it actually increases read load to downstream servers (A long discussion about that can be found in https://github.com/jacksontj/promxy/issues/2).

**Reply by megarajan** [reply]
thanks very much for the reply. I went through the MOTIVATION doc multiple times and got what you mentioned here, but  wanted to double confirm.
